corehq/apps/users/forms.py

```diff
@@ -1299,10 +1299,7 @@ def clean_role_id(self):
             and not self.couch_user.is_domain_admin(self.domain)
         )
         if not role_id:
-            if restricted_role_access:
-                raise forms.ValidationError(_("Please select a role"))
-            else:
-                return None
+            return None
```

Comment:
what happens if a non-admin access that page that has access to no roles available and `restricted_role_access` is true, submits the form? I think it would break. Currently the same will be true if someone manages to submit the form without a role, when `restricted_role_access` is true.

Reply:
Yeah, my guess is that the issue here is that `role_id` isn't available for some reason.  The validation seems correct - if a role _is_ selected, why isn't it showing up here?

Reply:
I believe its showing the error message when the form is being initialised and does not have a selected/initial value.